cluster/kubevirtci.sh

```diff
@@ -15,22 +15,39 @@
 export KUBEVIRT_PROVIDER=${KUBEVIRT_PROVIDER:-'k8s-1.17'}
 
 KUBEVIRTCI_VERSION='0b941ea5dc647d3aea6f6a6fff95563d6ce0445e'
-KUBEVIRTCI_PATH="${PWD}/_kubevirtci"
+KUBEVIRTCI_REPO='https://github.com/kubevirt/kubevirtci.git'
+KUBEVIRTCI_PATH=${KUBEVIRTCI_PATH:-"${PWD}/_kubevirtci/cluster-up/"}
+KUBEVIRTCI_PARENT_PATH="$(dirname "$KUBEVIRTCI_PATH")"
+
+function kubevirtci::_get_repo() {
+    git --git-dir ${KUBEVIRTCI_PARENT_PATH}/.git remote get-url origin
+}
+
+function kubevirtci::_get_version() {
+    git --git-dir ${KUBEVIRTCI_PARENT_PATH}/.git log --format="%H" -n 1
+}
 
 function kubevirtci::install() {
-    if [ ! -d ${KUBEVIRTCI_PATH} ]; then
-        git clone https://github.com/kubevirt/kubevirtci.git ${KUBEVIRTCI_PATH}
+    # Remove cloned kubevirtci repository if it does not match the requested one
+    if [ -d ${KUBEVIRTCI_PARENT_PATH} ]; then
+        if [ $(kubevirtci::_get_repo) != ${KUBEVIRTCI_REPO} -o $(kubevirtci::_get_version) != ${KUBEVIRTCI_VERSION} ]; then
+            rm -rf ${KUBEVIRTCI_PARENT_PATH}
+        fi
+    fi
+
+    if [ ! -d ${KUBEVIRTCI_PARENT_PATH} ]; then
+        git clone https://github.com/kubevirt/kubevirtci.git ${KUBEVIRTCI_PARENT_PATH}
         (
-            cd ${KUBEVIRTCI_PATH}
+            cd ${KUBEVIRTCI_PARENT_PATH}
             git checkout ${KUBEVIRTCI_VERSION}
         )
     fi
 }
 
 function kubevirtci::path() {
-    echo -n ${KUBEVIRTCI_PATH}
+    echo -n ${KUBEVIRTCI_PARENT_PATH}
```

Comment:
This is part of the confusion for me. kubevirtci::path does not return KUBEVIRTCI_PATH.

To make the distinction clear, would it make sense to rename this script to cluster.sh and all kubevirtci:: as well?

Overloading of the name kubevirtci is killing me here.